packages/not-next/src/hooks/usePublicClientByChainId.ts

```diff
@@ -0,0 +1,31 @@
+import { usePublicClient } from "wagmi";
+import { hardhat, mainnet, sepolia } from "viem/chains";
+
+export default function usePublicClientByChainId(id?: number) {
```

Comment:
this function should do what it say, get a publicClientByChainID, but it does more then that. But first lets start with getting a publicClientByChainID. To do this we only need to do
```
usePublicClient({
    chainId,
  });

```
I do not think wrapping this in another function make sense since it is fairly minimal. So lets consider what else usePublicClientByChainId is doing.
```
  const chainName = process.env.NEXT_PUBLIC_CHAIN_NAME!;
```
This is getting the chain that the store registry is deployed on. so maybe what we want this function to be called is usePublicClientForRegestry().